src/main/java/apoc/load/LoadXls.java

```diff
@@ -172,19 +179,47 @@ private Object convertArray(Object value) {
 
         private Object convertType(Object value) {
             if (nullValues.contains(value) || value == null) return null;
-//            if (!(value instanceof String)) return value;
-            switch (type) {
-                case STRING: return value.toString();
-                case INTEGER: return Util.toLong(value);
-                case FLOAT: return Util.toDouble(value);
-                case BOOLEAN: return Util.toBoolean(value);
-                case NULL: return null;
-                case LIST: return Arrays.stream(arrayPattern.split(value.toString())).map(this::convertType).collect(Collectors.toList());
-                default: return value;
+            switch (type) { // Destination Type
+                case STRING:
+                    if (value instanceof TemporalAccessor && !dateFormat.isEmpty()) {
+                        return dateFormat((TemporalAccessor) value, dateFormat);
+                    } else {
+                        return value.toString();
+                    }
+                case INTEGER:
+                    return Util.toLong(value);
+                case FLOAT:
+                    return Util.toDouble(value);
+                case BOOLEAN:
+                    return Util.toBoolean(value);
+                case NULL:
+                    return null;
+                case LIST:
+                    return Arrays.stream(arrayPattern.split(value.toString())).map(this::convertType).collect(Collectors.toList());
+                case DATE:
+                    return dateParse(value.toString(), LocalDate.class, dateParse);
+                case DATE_TIME:
+                    return dateParse(value.toString(), ZonedDateTime.class, dateParse);
+                case LOCAL_DATE_TIME:
+                    return dateParse(value.toString(), LocalDateTime.class, dateParse);
+                case LOCAL_TIME:
+                    return dateParse(value.toString(), LocalTime.class, dateParse);
+                case TIME:
+                    return dateParse(value.toString(), OffsetTime.class, dateParse);
+                case DURATION:
+                    return durationParse(value.toString());
+                default:
+                    return value;
             }
         }
     }
 
+    private static String[] convertFormat(Object value) {
+        if (value == null) return null;
+        List<String> strings = (List<String>) value;
```

Comment:
Manca il pezzo dove controlli se è un instance of di List<String> e nel caso bisogna ritornare un'eccezione con un messaggio appropriato